implementation/test/VendingMachineTest.js

```diff
@@ -443,6 +444,31 @@ describe("VendingMachine", async function() {
         expect(eventList[0].returnValues._digest).to.equal(sighash)
       })
 
+      it("returns true on success", async () => {
```

Comment:
Couple of things here:
- This block describes `RedemptionScript`, but as far as I can see, you're not actually returning true on success for `RedemptionScript`?
- This is really a test of `TBTCToken`, not `RedemptionScript`, right?

Same note below for `TBTCDepositToken` vs `FundingScript`.

Reply:
> This is really a test of TBTCToken 

Yes In a way. In order to test `approveAndCall` specifically, we would still have to go through `receiveApproval`, and this is implemented in our funding/redemption scripts. I can see how it's not ideal. 

The tests themselves are testing `approveAndCall` though.
`returns true on success` -> `approveAndCall returns true on success`?

Reply:
Sorry, I should have made my point clearer: this test is entirely a test of `TBTCToken`, so it feels like it should live in `TBTCTokenTest`. Similarly, the TDT one should live in `TBTCDepositTokenTest`. It feels fine to wrap these up into one `TokenTest` file or similar, but putting them in unrelated tests is not ideal.

In the interest of keeping things moving I'm going to merge this as-is, since it implements the underlying functionality we're looking for---but let's strive to make the changes quickly unless there was a good reason to have it in the current structure.